### ISSUE TITLE
[QUERY] Issue Retrieving Transactions with Microsoft.Azure.Management.Billing

### ISSUE THREAD

**Issue by DishanH** [Azure/azure-sdk-for-net#37886]
### Library name and version

Microsoft.Azure.Management.Billing

### Query/Question

File: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/billing/Microsoft.Azure.Management.Billing/tests/ScenarioTests/TransactionsOperationsTest.cs

Line no 36:
```
// Get the transactions
var transactions = billingMgmtClient.Transactions.ListByInvoice(BillingAccountName, InvoiceId);
```

I am experiencing an issue while attempting to retrieve transactions using Microsoft.Azure.Management.Billing for a particular account. The transactions and account queries are returning empty results, whereas the invoice query returns a full result.

This specific subscription is registered under both the Legacy MOSP (Microsoft Online Subscription Program) contract and the MCA (Microsoft Customer Agreement) modern contract. The billing account ID format for this account is 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx,' and it has multiple subscriptions attached to it. Despite trying, I failed to retrieve transaction and billing account information for this account. However, I have successfully retrieved invoice information using the below code.

Success
```
var billingClient = new BillingManagementClient(serviceCreds);
billingClient.SubscriptionId = "xxxxxxxx-cccc-cccc-cccc-xxxxxxxxxxxx";
var invoice = billingClient.Invoices.GetById("E0500O0000");
```
Failed (Empty result)
`var content = billingClient.Transactions.ListByInvoice(billingAccountName: "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", invoice.Name);`

On the other hand, I also have another account under the 'Microsoft Customer Agreement,' but with a different account ID format: 'yyyyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxx_2019-05-31.' Interestingly, I can successfully retrieve transaction data for this account using the same code."

### Environment

Windows 11
Visual Studio Code 1.80.1

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by DishanH** [comment]
@ArthurMa1978 Thank you for your response. It appears that the new SDK does not directly support retrieving Azure **invoices** or **Transactions**. Unfortunately, I couldn't find any documentation or sample related to this. Could you kindly guide me to the appropriate resources where I can find more information on this matter? Your assistance is much appreciated.

**Comment by ArthurMa1978** [comment]
It's interesting, the Track 2 sdk builds on the new api version [2021-10-01](https://github.com/Azure/azure-rest-api-specs/tree/main/specification/billing/resource-manager/Microsoft.Billing/stable/2021-10-01) but doesn't have the feature you need.
And with your next question of REST, that's all service issue which unfortunately we can't do any further help.  This repository is focused on the Azure SDK for .NET - we're unable to assist with other Azure issues, so your best path forward would be to [open an Azure support request](https://docs.microsoft.com/azure/azure-portal/supportability/how-to-create-azure-support-request).

**Comment by DishanH** [comment]
@ArthurMa1978 Yes, that was my first attempt. I currently have a service request (#2307190040010995) in progress. They asked me to create a GitHub support request because they don't have expert knowledge on this.

**Comment by ArthurMa1978** [comment]
> 

We need some time to explore the feasibility of launching a new SDK that supports both the old functions and old functions.

**Comment by ArthurMa1978** [comment]
Open https://github.com/Azure/azure-sdk-for-net/issues/40509 task to track this request.
